Describe changes in bulk (#10)
bioconda/bioconda-docs: source/developer/bulk.rst

Anaconda.** As such, only the bioconda core team has the ability to push to
this branch.

Interacting with the bulk branch
--------------------------------

When changes are made on the bulk branch (committed and pushed), the CI system
decides whether it will run its jobs or not.

* If any of the pushed commit messages contains the substring ``[ci run]`` the CI jobs are executed.
* If not, no CI jobs are executed.

The reason for this behavior is that we want to avoid race conditions caused by multiple CI jobs
spawned from different commits to be exectuted at the same time.

In order to simplify interactions with the bulk CI, bioconda-utils offers therefore
some dedicated subcommands:

* **bulk-commit**: ``bioconda-utils bulk-commit <message>`` commits the changes on your 
  local clone of https://github.com/bioconda/bioconda-recipes to the bulk branch while marking the commit
  as being eligible for a CI run (by automatically prefixing the message with ``[ci run]``).
  The **bulk-commit** subcommand does not push the commit. This enables you to do multiple fine-grained commits

>   and pus them in one pass via a subsequence ``git push`` that triggers a single CI run.

Reviewer: ```suggestion
  and push them in one pass via a subsequent ``git push`` that triggers a single CI run.
```

* **bulk-trigger-ci**: ``bioconda-utils bulk-trigger-ci`` creates an empty commit that is 
  immediately pushed automatically to the bulk branch, thereby triggering a CI run. This can be used
  to restart the CI run in case all of the previous runs are finished without build failures but there 

>   are still packages that need to be build (and haven't been before because the job runtime limits were

Reviewer: ```suggestion
  are still packages that need to be built (and haven't been before because the job runtime limits were
```

   There may be a few remaining conflicts to fix; in all cases you should
   prefer what's on the master branch.

5. Run ``bioconda-utils update-pinnings`` in
   the bulk branch. This will go through all the pinnings, figure out what
   recipes they're used with, and bump the recipes' build numbers
   appropriately. Then, **bulk-commit** and push the changes.

6. Once the CI run has finished, inspect all build failures (see :ref:`handling-build-failues`).
   For each failure, decide whether the recipe shall be skiplisted or whether you would like to fix it.
   In general it is advisable to fix all libraries on which many recipes depend and anything else
   that is obvious and easy. For the rest, mark the recipes as skiplisted in the build failure file.

>    It will be ignored by subsequence CI runs and put into a table in the bioconda-recipes wiki.

Reviewer: ```suggestion
   It will be ignored by subsequent CI runs and put into a table in the bioconda-recipes wiki.
```
